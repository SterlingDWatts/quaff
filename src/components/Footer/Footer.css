.Footer {
  background-color: rgb(78, 13, 58);
  box-shadow: 0 3px 6px rgba(0, 0, 0, 0.16), 0 3px 6px rgba(0, 0, 0, 0.23);
}

.Footer__container {
  padding: 16px;
  width: 100%;
  max-width: 944px;
  font-size: 24px;
  min-height: 56px;
  margin: auto;
}
