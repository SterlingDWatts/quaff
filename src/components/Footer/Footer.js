import React, { Component } from "react";
import QuizContext from "../../contexts/QuizContext";
import "./Footer.css";

class Footer extends Component {
  static contextType = QuizContext;

  renderProgress = () => {
    const { progress, numQuestions, numCorrect } = this.context;
    return (
      <div className="Footer__progress">
        <div>Material Covered: {`${progress}/${numQuestions}`}</div>
        <div>Correct: {`${numCorrect}/${numQuestions}`}</div>
      </div>
    );
  };

  render() {
    const progress = this.context.quiz.length > 1 ? this.renderProgress() : " ";
    return (
      <div className="Footer">
        <div className="Footer__container">{progress}</div>
      </div>
    );
  }
}

export default Footer;
