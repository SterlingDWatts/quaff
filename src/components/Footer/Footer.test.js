import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import Footer from "./Footer";

describe("Footer Component", () => {
  it("render without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <BrowserRouter>
        <Footer />
      </BrowserRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("render the UI as expected", () => {
    const tree = renderer.create(
      <BrowserRouter>
        <Footer />
      </BrowserRouter>
    );
    expect(tree).toMatchSnapshot();
  });
});
