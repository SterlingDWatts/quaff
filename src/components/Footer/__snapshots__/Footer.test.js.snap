// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component render the UI as expected 1`] = `
<div
  className="Footer"
>
  <div
    className="Footer__container"
  >
     
  </div>
</div>
`;
